### ISSUE TITLE
When uploading the same file as the file attached to the ticket, the file is moved to the dmsf directory.

### ISSUE THREAD

**Issue by sadakei2001** [danmunn/redmine_dmsf#1081]
When uploading the same file as the file attached to the ticket on the "classic" upload screen, the file is moved to the dmsf directory.

Environment:
 Redmine 3.4.6
 redmine_dmsf 1.6.1
 
 Redmine 4.1.0
 redmine_dmsf 2.4.2

On the classic upload screen,
A temporary file is created in "~/files/yyyy/mm" directory, and it moves to ”/files/dmsf/yyyy/mm" on commit.

Suppose that the ticket attached is uploaded from the classic screen.
In this case, the file with the ticket attached is regarded as a temporary file for dmsf upload by the file deduplication function(in Redmine 3.4 or late,), and it is moved to ”/files/dmsf/yyyy/mm" when committing.

Log when attaching a ticket

```
Started POST "/redmine/uploads.js?attachment_id=1&filename=dmsftestfile_20200131.xlsx&content_type=application%2Fvnd.openxmlformats-officedocument.spreadsheetml.sheet" for 127.0.0.1 at 2020-01-31 13:10:02 +0900
Processing by AttachmentsController#upload as JS
  Parameters: {"attachment_id"=>"1", "filename"=>"dmsftestfile_20200131.xlsx", "content_type"=>"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"}
  Current user: admin (id=1)
Saving attachment '/opt/redmine/files/2020/01/200131131002_dmsftestfile_20200131.xlsx' (7859 bytes)
```

Log when uploading dmsf by "classic upload"

```
_Started POST "/redmine/uploads.js?attachment_id=1&filename=dmsftestfile_20200131.xlsx&content_type=application%2Fvnd.openxmlformats-officedocument.spreadsheetml.sheet" for 127.0.0.1 at 2020-01-31 13:10:28 +0900
Processing by AttachmentsController#upload as JS
  Parameters: {"attachment_id"=>"1", "filename"=>"dmsftestfile_20200131.xlsx", "content_type"=>"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"}
  Current user: admin (id=1)
Saving attachment '/opt/redmine/files/2020/01/200131131029_dmsftestfile_20200131.xlsx' (7859 bytes)_
```

Actually, no new file is created and "200131131002_dmsftestfile_20200131.xlsx" moves to ”/files/dmsf/yyyy/mm".

**Comment by picman** [comment]
Not sure whether I fully understand your issue. However, one of the features of DMSF is that if a file is being uploaded and a file of the same filename already exists, no new file is created but just a new revision of the existing file is created. Isn't it the case?

**Comment by sadakei2001** [comment]
In this case, it is a problem when uploading the same file attached to Redmine “issue” to DMSF.
It's OK to save a new revision to the DMSF.
However, clicking the attachment link in Redmine's "issue" results in a 404 error.

**Comment by danaivehr** [comment]
Hi! I can confirm this behavior.

Steps to reproduce:
* Get some file
* Edit any issue and attach this file to the issue (core Redmine attachment)
* Try to download file - all is ok with it
* Now upload **the same file** to DMSF
* Try to download file from DMSF - all is ok wit it
* Now get back to that issue with first attachment and try to download that file again - 403 error

If you look into redmine "files" folder on disk, it seems that file just disappears from original files/YEAR/MONTH/ location after being uploaded to DMSF

**Comment by danaivehr** [comment]
@picman You use `a = Attachment.find_by_token(uploaded_file[:token])` after upload, which return you an existing file already attached in redmine in this case.

It seems that Redmine logic for the same files uploads conflicts with your plugin logic - you expect the 'temporary' file in this case, but redmine gives you already matched attachment. Then you deal with it like with temporary file - move it in dmsf subfolder.

Unfortunately i'm not familiar with Ruby, so cannot get a workaround for it.

**Comment by picman** [comment]
Attachment:

```Saving attachment '/opt/redmine/files/2020/01/200131131002_dmsftestfile_20200131.xlsx'```

DMS:

```Saving attachment '/opt/redmine/files/2020/01/200131131029_dmsftestfile_20200131.xlsx'```

=> The problem is, that you have set the same destination folder for _**Files**_ and for **_Documents_**. DMS supposes that it has it's own folder, e.g. _/opt/redmine/dmsf_. Change your DMSF plugin's settings.

**Comment by picman** [comment]
I've just tested exactly the same procedure with _punk8.jpg_ file. After updating an issue and uploading the same file into DMS I still see both files in their independent locations:
```
$ find dmsf/2020/06/ -name "*punk8.jpg*"
dmsf/2020/06/200610111656_221585_punk8.jpg

$ find files/2020/06/ -name "*punk8.jpg*"
files/2020/06/200610131632_punk8.jpg
```

**Comment by picman** [comment]
No DB changes needed. Just to change the path in the settings and move the files.

**Comment by danaivehr** [comment]
Thank you!
I can confirm that moving dmsf directory out of ./files resolves this issue!

**Comment by danaivehr** [comment]
> Thank you!
> I can confirm that moving dmsf directory out of ./files resolves this issue!

No, wait. Seems to be no a straight forward bug reproduction - issue still present.
I'll try too find exact steps to reproduce it.

**Comment by pskillen** [comment]
I'm seeing the same bug with the same steps to reproduce (from the comment on 4 Jun). Both when storing files under "files/dmsf" and just under "dmsf".

Environment:
Redmine 4.1 (from official Docker image)
DMSF 2.4.4

PS: thanks @danaivehr for finding the steps to reproduce - I've known _something_ was wrong for ~2 years and I couldn't reproduce it reliably

**Comment by jlg29387** [comment]
I have the same bug.

I move dmsf directory out of ./files specifically to the route 'redmine/htdocs/dmsf' and I changed the path in the settings of the plugin from admin panel. 

The problem keeps happening. Any solution? **Is it necessary to change any of the plugin files too?** 

My steps are the same  https://github.com/danmunn/redmine_dmsf/issues/1081#issuecomment-639130446

**Comment by picman** [comment]
Could you re-test it with the latest release 2.4.5, please?

**Comment by jlg29387** [comment]
When updating to the last realise, seems works for me. Thanks

**Comment by picman** [comment]
Fixed in #1185.
